core/domain/question_services.py

```diff
@@ -117,6 +119,23 @@ def get_question_by_id(question_id):
         return None
 
 
+def get_questions_by_ids(question_ids):
+    """Returns a list of domain objects representing questions.
+
+    Args:
+        question_ids: list(str). List of question ids.
+
+    Returns:
+        list(Question) or None. A list of domain objects representing questions
+        with the given ids or None when the id is not valid.
+    """
+    question_model_list = question_models.QuestionModel.get_multi(question_ids)
+    questions = []
+    for question_model in question_model_list:
+        questions.append((get_question_from_model(question_model)).to_dict())
```

Comment:
I don't think you need the inner parenthesis.

Also, why are you doing to_dict() if you're returning a list of domain objects? to_dict() converts to a Python dict.

Reply:
Updated.
I need to convert them to dicts to make it JSON serialisable.

Reply:
OK -- FYI, that conversion should only happen in the controller, then, just at the last minute before the data is passed to the frontend.

Reply:
but wouldn't that create another for loop?
I will have to loop through all the objects of the list and apply to_dict method to each object.
I am assuming that better data structure and usability the reasons for the trade off of efficiency(which is insignificant as runtime is approximately O(n) for both the cases). Please let me know If I am missing something here.

Reply:
That's right -- we use domain objects throughout the codebase as much as we can, and only convert to raw values at the boundaries of the codebase. This is due to the increased structure, checks, etc. provided by domain objects.

Reply:
Thank you for your explanation. 
Updated in the latest push.